controllers/authController.js

```diff
@@ -0,0 +1,77 @@
+const { db } = require("../models");
+const UserService = require("../services/UserService");
+const userService = new UserService(db);
+const { hashPassword, verifyPassword } = require("../utilities/hashing");
+const RoleService = require("../services/RoleServices");
+const roleService = new RoleService(db);
+const { generateToken } = require("../utilities/jwt");
+
+async function register(req, res) {
+  const { firstName, lastName, username, email, password } = req.body;
+  const { salt, hashedPassword } = await hashPassword(password);
+
+  //Created a samll RoleService.
+  const role = await roleService.getOneRole("user");
+
+  //creating the inital user
+  const user = await userService.create({
+    firstName: firstName,
+    lastName: lastName,
+    username,
+    displayName: username,
+    email,
+    hashedPassword,
+    salt,
+    roleId: role.id,
+  });
```

Comment:
Not really a problem just making it look pretty
```suggestion
  const user = await userService.create({
    firstName,
    lastName,
    username,
    displayName: username,
    email,
    hashedPassword,
    salt,
    roleId: role.id,
  });
```

Reply:
Added